public/app/features/explore/TraceView.tsx

```diff
@@ -0,0 +1,247 @@
+import {
+  DetailState,
+  KeyValuePair,
+  Link,
+  Log,
+  Span,
+  // SpanData,
+  // SpanReference,
+  Trace,
+  TraceTimelineViewer,
+  TTraceTimeline,
+  UIElementsContext,
+  ViewRangeTimeUpdate,
+  transformTraceData,
+  SpanData,
+  TraceData,
+} from '@jaegertracing/jaeger-ui-components';
+import React, { useState } from 'react';
+
+type Props = {
+  trace: TraceData & { spans: SpanData[] };
+};
+
+export function TraceView(props: Props) {
+  /**
+   * Track whether details are open per span.
+   */
+  const [detailStates, setDetailStates] = useState(new Map<string, DetailState>());
```

Comment:
In jaeger-ui all this was implemented as redux actions/reducers so it was taken out and we have to reimplement it here. Most of the handlers code is taken from their reducers though.